docs/source/quickstart.md

```diff
@@ -1,88 +1,271 @@
 # Quickstart
 
-## How does it work?
+TRL is a comprehensive library for post-training foundation models using techniques like Supervised Fine-Tuning (SFT), Direct Preference Optimization (DPO), and Group Relative Policy Optimization (GRPO).
 
-Fine-tuning a language model via PPO consists of roughly three steps:
+## Installation
 
-1. **Rollout**: The language model generates a response or continuation based on a query which could be the start of a sentence.
-2. **Evaluation**: The query and response are evaluated with a function, model, human feedback, or some combination of them. The important thing is that this process should yield a scalar value for each query/response pair. The optimization will aim at maximizing this value.
-3. **Optimization**: This is the most complex part. In the optimisation step the query/response pairs are used to calculate the log-probabilities of the tokens in the sequences. This is done with the model that is trained and a reference model, which is usually the pre-trained model before fine-tuning. The KL-divergence between the two outputs is used as an additional reward signal to make sure the generated responses don't deviate too far from the reference language model. The active language model is then trained with PPO.
+```bash
+pip install trl
+```
+
+## Quick Examples
+
+Get started instantly with TRL's most popular trainers. Each example uses compact models for quick experimentation.
+
+<div class="trainer-toggle">
+
+### SFT Trainer
+
+**Supervised Fine-Tuning**
+
+```python
+from trl import SFTTrainer, SFTConfig
+from datasets import load_dataset
+
+config = SFTConfig(
+    output_dir="./qwen-chat-model",
+    per_device_train_batch_size=8,
+    gradient_checkpointing=True, 
+    gradient_accumulation_steps=1,
+    learning_rate=2e-5,
+    logging_steps=10,
+)
+
+trainer = SFTTrainer(
+    model="Qwen/Qwen2.5-0.5B",
+    train_dataset=load_dataset("trl-lib/Capybara", split="train[:1000]"),
+)
+trainer.train()
+trainer.save_model()
+```
 
-The full process is illustrated in the following figure:
-<img src="https://huggingface.co/datasets/trl-lib/documentation-images/resolve/main/trl_overview.png"/>
+### DPO Trainer  
 
-## Minimal example
+**Direct Preference Optimization**
 
-The following code illustrates the steps above. 
+```python
+from trl import DPOTrainer, DPOConfig
+from datasets import load_dataset
+
+# DPO requires an SFT model as base
+config = DPOConfig(
+    output_dir="./qwen-aligned",
+    per_device_train_batch_size=8,
+    gradient_checkpointing=True,
+    gradient_accumulation_steps=1,
+    learning_rate=1e-6,
+    beta=0.1,
+    logging_steps=10,
+)
+
+trainer = DPOTrainer(
+    model="./qwen-chat-model",  # Use your SFT model
```

Comment:
nit: let's make the examples work OOB with copy-paste by using a public chat model
```suggestion
    model="Qwen/Qwen2.5-0.5B-Instruct",  # The model to optimize
```